flocker/node/agents/blockdevice.py

```diff
@@ -408,6 +419,34 @@ def run(self, deployer):
         return succeed(None)
 
 
+def _ensure_no_filesystem(device):
+    """
+    Raises an error if there's already a filesystem on ``device``.
+
+    :raises: ``FilesystemExists`` if there is already a filesystem on
+        ``device``.
+    :return: ``None``
+    """
+    try:
+        check_output(
+            [b"blkid", b"-p", b"-u", b"filesystem", device.path],
+            stderr=STDOUT,
+        )
+    except CalledProcessError as e:
+        # According to the man page:
+        #   the specified token was not found, or no (specified) devices
+        #   could be identified
+        #
+        # Experimentation shows that there is no output in the case of the
+        # former, and an error printed to stderr in the case of the
+        # latter.
+        if e.returncode == 2 and not e.output:
+            # There is no filesystem on this device.
+            return
+        raise
```

Comment:
It'd be nice if this case could be tested .... but Itamar and I couldn't see a reliable way to trigger the other error cases (at least not without screwing with the arguments).  Maybe just leave a comment noting it's untested, possibly linking to a follow-up issue for addressing this somehow (maybe by switching to a blkid library with a test double with support for easily exercising error paths).